corehq/apps/ota/admin.py

```diff
@@ -35,8 +35,16 @@ def clean(self):
                 raise ValidationError("You must specify both {} and {}."
                                       .format(version_min, version_max))
 
-        if data['cc_all_versions'] and data['app_all_versions']:
-            raise ValidationError("You cannot apply a measure to all cc and app versions.")
+        application_measures = ['app_reinstall_ota', 'app_reinstall_local', 'app_update']
+        commcare_measures = ['cc_reinstall', 'cc_update']
+        if data['measure'] in application_measures:
+            if data['app_all_versions']:
+                raise ValidationError('App related measures must specify an app version range')
+        elif data['measure'] in commcare_measures:
+            if data['cc_all_versions']:
+                raise ValidationError('Commcare related measures must specify a commcare version range')
+        else:
+            raise ValidationError("I don't recognize this measure so I can't validate it.")
```

Comment:
I think these are only shown internally, but it might be useful to provide a list of acceptable measures in this message.

Reply:
Changed the message and switched to an `AssertionError` to make it clear that this is a code error, not a user error